keda/templates/metrics-server/clusterrolebinding.yaml

```diff
@@ -57,6 +59,6 @@ roleRef:
   name: {{ .Values.operator.name }}-external-metrics-reader
 subjects:
 - kind: ServiceAccount
-  name: horizontal-pod-autoscaler
-  namespace: kube-system
```

Comment:
to be honest I wasn't sure about this one. I did a brief research by checking who is this service account and I didn't find the service account called `horizontal-pod-autoscaler` in `kube-system` ns in any k8s cluster I have running so I thought it's a obsolete thing. However, I found it now on kind and k3d. It wasn't present in GCP and AWS that used the "virtualized control planes".

To my knowledge hpa controller runs within the kube-controller-manager which doesn't use such service account. But I learned that there is a flag for `kube-controller-manager` process called `--use-service-account-credentials` that acutally splits the rbac for individual controllers contained in the `kube-controller-manager`

there is also that clusterrole that has
```
k describe clusterrole system:kube-controller-manager
  *.*                                        []                 []                         [list watch]
```

that might capture this, but just to be sure, I am adding this back.

Reply:
yeah, we need to keep it because depending on the control plane config, this can be required or not and I think that it's worth adding it always as it's something quite in depth for the average (managed k8s) user: https://docs.datadoghq.com/security/default_rules/cis-kubernetes-1.5.1-1.3.3/